src/main/kotlin/io/github/detekt/gradle/DetektKotlinCompilerPlugin.kt

```diff
@@ -43,3 +48,14 @@ class DetektKotlinCompilerPlugin : KotlinCompilerPluginSupportPlugin {
 
     override fun isApplicable(kotlinCompilation: KotlinCompilation<*>): Boolean = true
 }
+
+internal fun ConfigurableFileCollection.toDigest(): String {
+    val concatenatedConfig = this
+        .filter { it.isFile }
+        .map(File::readBytes)
+        .fold(byteArrayOf()) { acc, file -> acc + file }
+
+    return Base64.getEncoder().encodeToString(
+        MessageDigest.getInstance("SHA-256").digest(concatenatedConfig)
```

Comment:
We could use `DigestInputStream` or something similar so we don't need to copy all the file in memory to compute the sha. But thinking about what detekt and the Compiler does this is clearly nothing so 🤐.